crates/red_knot_python_semantic/src/semantic_index/builder.rs

```diff
@@ -519,7 +519,6 @@ impl<'db> SemanticIndexBuilder<'db> {
     }
 
     /// Records a visibility constraint by applying it to all live bindings and declarations.
-    #[must_use = "A visibility constraint must always be negated after it is added"]
```

Comment:
Removed this because it is not true, and this PR is an example of where it is not true; the "other branch" for an assert statement is an exception, which exits the scope, so there is no other branch in which the visibility constraint should be negated.

Reply:
Thanks for the correction!